charts/argo-cd/templates/argocd-repo-server/repository-credentials-secret.yaml

```diff
@@ -1,13 +1,18 @@
-{{- if .Values.configs.repositoryCredentials }}
+{{- if .Values.configs.repositoryTemplates }}
 apiVersion: v1
-kind: Secret
-metadata:
-  name: argocd-repository-credentials
-  labels:
-    {{- include "argo-cd.labels" (dict "context" . "component" .Values.server.name "name" "secret") | nindent 4 }}
-type: Opaque
-data:
-{{- range $key, $value := .Values.configs.repositoryCredentials }}
-  {{ $key }}: {{ $value | b64enc }}
+kind: List
```

Comment:
Can we prevent use of "kind: List" and prefer 
```yaml
{{- range foo}}
---
apiVersion: v1
kind: Secret
metadata:
# (..)
{{- end }}
```
There were bugs in the past in helm lint. Also I am unsure if this is fulls supported when deploying this chart through another ArgoCD instance

Reply:
Great. I will change.

Reply:
We need the `---` on the first line inside the range

Reply:
Done. Thanks the help. I'm initiating the contribution with communities.

Reply:
👍 oh and the if around the range can be omitted. If `configs.repositories` is empty, range will not render things.

Reply:
Excellent.